floor_generator/lib/writer/database_writer.dart

```diff
@@ -94,24 +94,50 @@ class DatabaseWriter implements Writer {
       ..name = 'migrations'
       ..type = refer('List<Migration>'));
 
+    final onConfigureParameter = Parameter((builder) => builder
+      ..name = 'onConfigure'
+      ..named = true
+      ..type = refer('sqflite.OnDatabaseConfigureFn'));
+
+    final onCreateParameter = Parameter((builder) => builder
+      ..name = 'onCreate'
+      ..named = true
+      ..type = refer('sqflite.OnDatabaseCreateFn'));
+
+    final onUpgradeParameter = Parameter((builder) => builder
+      ..name = 'onUpgrade'
+      ..named = true
+      ..type = refer('sqflite.OnDatabaseVersionChangeFn'));
+
     return Method((builder) => builder
       ..name = 'open'
       ..returns = refer('Future<sqflite.Database>')
       ..modifier = MethodModifier.async
       ..requiredParameters.addAll([nameParameter, migrationsParameter])
+      ..optionalParameters
+          .addAll([onConfigureParameter, onCreateParameter, onUpgradeParameter])
```

Comment:
Please add a comma after `onUpgradeParameter` and reformat the file in order to have a proper line break.

```suggestion
          .addAll([onConfigureParameter, onCreateParameter, onUpgradeParameter,])
```